pkg/api/steve/supportconfigs/generator.go

```diff
@@ -0,0 +1,169 @@
+	unauthenticatedError = "unauthenticaed"
+	forbiddenError       = "forbidden"
+)
+
+type GeneratorHandler struct {
+	ConfigMaps           v1.ConfigMapInterface
+	SubjectAccessReviews authv1.SubjectAccessReviewInterface
+	ConfigHandler        supportConfigHandler
+}
+
+func NewGeneratorHandler(scaledContext *config.ScaledContext) GeneratorHandler {
+	return GeneratorHandler{
+		ConfigMaps:           scaledContext.Core.ConfigMaps(cspAdapterNamespace),
+		SubjectAccessReviews: scaledContext.K8sClient.AuthorizationV1().SubjectAccessReviews(),
+		ConfigHandler:        supportConfigHandler{},
+	}
+}
+
+func (h *GeneratorHandler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
+	err := h.authorize(request)
+	if err != nil {
+		switch err.Error() {
+		case unauthenticatedError:
+			writer.WriteHeader(http.StatusUnauthorized)
+			_, _ = writer.Write([]byte(err.Error()))
+		case forbiddenError:
+			writer.WriteHeader(http.StatusForbidden)
+			_, _ = writer.Write([]byte(err.Error()))
+		default:
+			writer.WriteHeader(http.StatusInternalServerError)
+			logrus.Errorf("Failed to authorize user with error: %s", err.Error())
+		}
+		return
+	}
+	logrus.Infof("Generating supportconfig")
+	archive, err := h.generateSupportConfig()
+	logrus.Infof("Done Generating supportconfig")
+	if err != nil {
+		logrus.Errorf("Error when generating support config %v", err)
+		writer.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	writer.Header().Set("Content-Type", tarContentType)
+	writer.Header().Set("Content-Disposition", "filename=\"supportConfig.tar\"")
+	n, err := io.Copy(writer, archive)
+	logrus.Infof("wrote %v bytes in archive response", n)
+}
+
+// authorize checks to see if the user can get the non-resource url of Endpoint. If not, returns an error, otherwise nil
+func (h *GeneratorHandler) authorize(r *http.Request) error {
+	userInfo, ok := request.UserFrom(r.Context())
+	if !ok {
+		return fmt.Errorf(unauthenticatedError)
+	}
+	response, err := h.SubjectAccessReviews.Create(r.Context(), &authzv1.SubjectAccessReview{
+		Spec: authzv1.SubjectAccessReviewSpec{
+			// This auth method prevents collision with future k8s resources and allows this to be added to custom rules
+			NonResourceAttributes: &authzv1.NonResourceAttributes{
+				Verb: "get",
+				Path: Endpoint,
+			},
+			User:   userInfo.GetName(),
+			Groups: userInfo.GetGroups(),
+			Extra:  convertExtra(userInfo.GetExtra()),
+			UID:    userInfo.GetUID(),
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create sar %s", err)
+	}
+	if !response.Status.Allowed {
+		return fmt.Errorf(forbiddenError)
+	}
+	return nil
+}
+
+// generateSupportConfig produces an io.Reader with the supportConfig ready to be returned using a http.ResponseWriter
+// Returns an err if it can't get the support config (note: does not return an err if it can't find the csp config)
+func (h *GeneratorHandler) generateSupportConfig() (io.Reader, error) {
```

Comment:
Long term we should include pod logs in here. Should be able to read in with k8s client. Not necessary right now.